test/spec/search.js

```diff
@@ -0,0 +1,376 @@
+    yield browser
+      .url('/')
+      .click('#school-name h1 a')
+      .setValue('#name-school', 'col')
+      .click('#search-submit');
+
+    var result = yield getResultCount();
+    assert.equal(result, results.resultCounts.allCol);
+  });
+
+  it('should contain no results for a nonsense word', function*() {
+    yield browser
+      .url('/')
+      .click('#school-name h1 a')
+      .setValue('#name-school', 'nons%ense wo@d#')
+      .click('#search-submit');
+
+    var result = yield getResultCount();
+    assert.equal(result, '0');
+  });
+
+  /*
+    Search By Location 
+  */
+  it('should contain all schools in Delaware if selected', function*() {
+    yield browser
+      .url('/')
+      .click('#school-location h1 a')
+      .selectByValue('.multi-select_item .select-state', 'DE')
+      .click('#search-submit');
+
+    var result = yield getResultCount();
+    assert.equal(result, results.resultCounts.allInDelaware);
+  });
+
+  it('should contain all schools in DE & KY if selected', function*() {
+    yield browser
+      .url('/')
+      .click('#school-location h1 a')
+      .selectByValue('.multi-select_item .select-state', 'DE')
+      .click('button.button.button-add')
+      .selectByValue('.multi-select_item + .multi-select_item .select-state', 'KY')
+      .click('#search-submit');
+
+    var result = yield getResultCount();
+    assert.equal(result, results.resultCounts.allInDelawareKentucky);
+  });
+
+  it('should contain all schools in Far West region if selected', function*() {
+    yield browser
+      .url('/')
+      .click('#school-location h1 a')
+      .selectByValue('.multi-select_item .select-region', '8')
+      .click('#search-submit');
+
+    var result = yield getResultCount();
+    assert.equal(result, results.resultCounts.allInFarWest);
+  });
+
+  it('should contain all schools in Zip Code 10023', function*() {
+    yield browser
+      .url('/')
+      .click('#school-location h1 a')
+      .setValue('#zip-code', '10023')
+      .click('#search-submit');
+
+    var result = yield getResultCount();
+    assert.equal(result, results.resultCounts.allIn10023NoRadius);
+  });
+
+  it('should contain all schools within 5 miles of Zip Code 10023', function*() {
+    yield browser
+      .url('/')
+      .click('#school-location h1 a')
+      .setValue('#zip-code', '10023')
+      .setValue('#search-radius', '5')
+      .click('#search-submit');
+
+    var result = yield getResultCount();
+    assert.equal(result, results.resultCounts.allIn10023Radius5);
+  });
+
+  // CLARIFICATION: The spec says this should error, but it just returns 0 results?
+  // Returns an error if you define a radius + bogus zip.
+  it('should contain no results for zip code = 1089987987', function*() {
+    yield browser
+      .url('/')
+      .click('#school-location h1 a')
+      .setValue('#zip-code', '1089987987')
+      .click('#search-submit');
+
+    var result = yield getResultCount();
```

Comment:
'yield' is available in ES6 (use esnext option) or Mozilla JS extensions (use moz).